File: iosApp/iosApp/iosApp/Home/Render/ManifestBlockView.swift
import SwiftUI
import Shared

@ViewBuilder
func ManifestBlockView(_ block: ManifestBlock, onClassClick: @escaping (String) -> Void) -> some View {
    switch onEnum(of: block) {
    case .greeting(let b):
        GreetingBlock(block: b)
    case .hero(let b):
        HeroBlock(block: b)
    case .myClub(let b):
        MyClubBlock(block: b)
    case .classCarousel(let b):
        ClassCarouselBlock(block: b, onClassClick: onClassClick)
    case .myRewards(let b):
        MyRewardsBlock(block: b)
    case .myGoals(let b):
        MyGoalsBlock(block: b)
    case .promotion(let b):
        PromotionBlock(block: b)
    case .unknown:
        EmptyView()
    }
}

File: androidApp/src/main/kotlin/com/virginactive/android/ui/home/render/ManifestBlockView.kt
package com.virginactive.android.ui.home.render

import androidx.compose.runtime.Composable
import com.virginactive.shared.domain.home.ManifestBlock

@Composable
fun ManifestBlockView(
    block: ManifestBlock,
    onClassClick: (String) -> Unit,
) {
    when (block) {
        is ManifestBlock.Greeting -> GreetingBlock(block)
        is ManifestBlock.Hero -> HeroBlock(block)
        is ManifestBlock.MyClub -> MyClubBlock(block)
        is ManifestBlock.ClassCarousel -> ClassCarouselBlock(block, onClassClick)
        is ManifestBlock.MyRewards -> MyRewardsBlock(block)
        is ManifestBlock.MyGoals -> MyGoalsBlock(block)
        is ManifestBlock.Promotion -> PromotionBlock(block)
        is ManifestBlock.Unknown -> Unit
    }
}
